week6/Kjiw0n/mission/src/pages/Store/Store.jsx

```diff
@@ -0,0 +1,66 @@
+import { useNavigate, useParams } from "react-router-dom";
+
+import MenuItem from "../../components/MenuItem/MenuItem";
+
+import stores from "../../models/stores";
+import BackBtn from "../../components/BackBtn";
+import styled from "styled-components";
+import StoreDetailInfo from "../../components/Store/StoreDetailInfo";
+
+import "../Store/Store.css";
+
+const Store = () => {
+  const { storeId } = useParams();
+  const store = stores.find((store) => store.id.toString() === storeId);
+  const navigate = useNavigate();
+
+  if (!store) {
+    return <div>가게를 찾을 수 없어요 🥺</div>;
+  }
+
+  return (
+    <div>
+      <BackBtn onClick={() => navigate(-1)} />
+
+      <StoreDetailInfo />
+
+      <Line></Line>
+
+      <CategoryTitle>샐러드</CategoryTitle>
+
+      <MenuItemWrapper>
+        {store.menus.map((menu) => {
+          return <MenuItem key={menu.id} menu={menu} />;
+        })}
+      </MenuItemWrapper>
+    </div>
+  );
+};
+
+export default Store;
+
+const Line = styled.div`
+  width: 100%;
+  height: 1px;
+  background: #e5e8eb;
+
+  margin-bottom: 26px;
+`;
+
+const CategoryTitle = styled.h4`
+  color: #6b7684;
+  font-family: Pretendard;
+  font-size: 17px;
+  font-style: normal;
+  font-weight: 600;
+  line-height: normal;
+
+  margin: 0 0 11px 24px;
+`;
+
+const MenuItemWrapper = styled.div.attrs({
+  className: "MenuItemWrapper",
+})`
+  height: 380px;
+  overflow-y: auto;
+`;
```

Comment:
- attrs로 속성 부여할 수 있다는건 저도 첨알았네요

- 여기에 `::-webkit-scrollber`를 바로 썼을 때 안되는 이유는, `:`으로 붙이는 pseudo css들은 원래 그 자신에 붙어야 효력이 발휘됩니다.

- 즉, css로 해주신 것처럼 해당 className에 직접 붙어야 효과가 나타나는데, styled-components 안에 `::-webkit-scrollbar` 속성을 넣으면 하위 태그를 선택하는 것처럼 동작해서 동작이 안 되는 것이죠!!
```css
.MenuItemWrapper::-webkit-scrollbar {
  display: none;
}
```

- 따라서 어떤 css 안에서 그 자신을 선택하기 위해 앞에 `&` 선택자 붙여주면 자기 자신에 pseudo css를 붙일 수 있어요

```suggestion
const MenuItemWrapper = styled.div.attrs({
  className: "MenuItemWrapper",
})`
  height: 380px;
  overflow-y: auto;

  /* 스크롤바 숨김 */
  &::-webkit-scrollbar {
    display: none;
  }
`;
```

Reply:
아하.. 저 바보네요. 감사합니다 👍 👍